toplevel/vernac.ml

```diff
@@ -105,11 +105,52 @@ let pr_open_cur_subgoals () =
     Printer.pr_open_subgoals ~proof
   with Proof_global.NoCurrentProof -> Pp.str ""
 
+let prev_step = ref ""  (* output from the previous proof step of the proof, used for diff *)
```

Comment:
Use Summary.ref to get a reference which respects backtracking (it registers the ref with the system which resets things when you backtrack).

Reply:
If this PR is rebased on top of #6347 the summary hack is not needed as the toplevel has access to both proofs.

Reply:
I think you used the wrong PR number.

Reply:
When will #6374 be merged?  Is it basically complete but waiting for some other event before merging?  Are you suggesting that we hold this PR until 6374 is merged?  It sounds like 6374 would provide a cleaner interface for the diffs, but the changes to this PR to use 6374 would be quite limited.  I'm not accustomed to basing one PR on another--it seems like that might introduce more instability into my dev process.

Reply:
> When will #6374 be merged?

Pretty soon I hope as I have done some testing and it seems ready.

> Is it basically complete but waiting for some other event before merging?

Yup. It has quite a long dep chain. Unfortunately merging is not very agile in a project as complex as Coq.

> I'm not accustomed to basing one PR on another--it seems like that might introduce more instability into my dev process.

AFAICT you can just rebase on top of the PR branch instead of master, but if that seems tricky to do I suggest you wait for #6374 to be merged. Anyways, this PR needs a bit of work on other areas too.

Reply:
> Pretty soon I hope as I have done some testing and it seems ready.
> Unfortunately merging is not very agile in a project as complex as Coq.

Note that #6374 has a "needs: independent fix" label, so it is not on my radar.

Reply:
@maximedenes I didn't switch the label to `needs: testing` as both have the same effect. Will thou do it for accuracy.